### PR TITLE
[Merged by Bors] - chore: more cleanup of nolint simpNF

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Category/Profinite/Basic.lean [hunk 1]
```diff
@@ -145,6 +145,8 @@ def FintypeCat.toProfinite : FintypeCat ⥤ Profinite where
   obj A := Profinite.of A
   map f := ⟨f, by continuity⟩
 
+-- This lemma is generated by `simps` but we don't actually want the `@[simp]` attribute.
+-- https://github.com/leanprover-community/mathlib4/issues/18942
 attribute [nolint simpNF] FintypeCat.toProfinite_map_apply
```
Comment (b-mehta) [inline Mathlib/Topology/Category/Profinite/Basic.lean:150]: Might it be better to do `attribute [-simp] FintypeCat.toProfinite_map_apply` here? That way we can remove the `nolint simpNF`? I'm happy either way round.

File: Mathlib/Topology/ContinuousMap/Units.lean [hunk 2]
```diff
@@ -45,13 +45,11 @@ def unitsLift : C(X, Mˣ) ≃ C(X, M)ˣ where
   left_inv f := by ext; rfl
   right_inv f := by ext; rfl
 
--- Porting note: add manually because `simps` used `inv` and `simpNF` complained
```
Comment (b-mehta) [inline Mathlib/Topology/ContinuousMap/Units.lean:48]: This porting note looks like it matches #18942 exactly, perhaps we should add a link to that to the comment earlier (on line 28). And then it's not really a porting note any more!

**Review by b-mehta — APPROVED** [review]
bors d+

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean [hunk 3]
```diff
@@ -1094,13 +1094,15 @@ theorem bfamilyOfFamily_typein {ι} (f : ι → α) (i) :
     bfamilyOfFamily f (typein _ i) (typein_lt_type _ i) = f i :=
   bfamilyOfFamily'_typein _ f i
 
-@[simp, nolint simpNF] -- Porting note (#10959): simp cannot prove this
+-- See https://github.com/leanprover-community/batteries/issues/365 for the `nolint simpNF`.
+@[simp, nolint simpNF]
```
Comment (vihdzp) [inline Mathlib/SetTheory/Ordinal/Arithmetic.lean:1098]: If this isn't being used as a `simp` lemma, I think it's fine to remove this outright. This `BFamily` API has no reason to be used in any new theorems and is planned to be deprecated soon.

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean [hunk 4]
```diff
@@ -1487,7 +1489,8 @@ theorem sup_eq_bsup' {o : Ordinal.{u}} {ι} (r : ι → ι → Prop) [IsWellOrde
     (f : ∀ a < o, Ordinal.{max u v}) : sup.{_, v} (familyOfBFamily' r ho f) = bsup.{_, v} o f :=
   sup_eq_sup r _ ho _ f
 
-@[simp, nolint simpNF] -- Porting note (#10959): simp cannot prove this
+-- See https://github.com/leanprover-community/batteries/issues/365 for the `nolint simpNF`.
+@[simp, nolint simpNF]
```
Comment (vihdzp) [inline Mathlib/SetTheory/Ordinal/Arithmetic.lean:1493]: Same here.

**Comment by adomani** [comment]
Testing a new entry to the `PR summary` script.  For this PR, it would report the following:

---

<details><summary>Changes to technical debts</summary>

|Current number|Change|Type|
|-:|:-:|:-|
|5109|-20|porting notes|
|229|-11|disabled simpNF lints|

Current commit [651df49d2d](https://github.com/leanprover-community/mathlib4/commit/651df49d2d8c69f4316603922bd6a800e4bf4072)
Reference commit [6a85082cb2](https://github.com/leanprover-community/mathlib4/commit/6a85082cb2ae80ff81a30ecea04bcbf258d51402)

You can run this locally as
```
./scripts/technical-debt-metrics.sh pr_summary
```
</details>

**Comment by kim-em** [comment]
bors merge

**Comment by kim-em** [comment]
bors merge

File: Mathlib/Topology/Category/Profinite/Basic.lean (hunk 1 above)
Comment (kim-em) [reply Mathlib/Topology/Category/Profinite/Basic.lean]: @b-mehta this doesn't actually work. `attribute [-simp]` is only file local.

Comment (kim-em) [reply Mathlib/Topology/Category/Profinite/Basic.lean:150]: I've reverted.

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**